src/calculator.js

```diff
@@ -0,0 +1,22 @@
+import { add, divide, multiply, subtract } from './utils/calculation.js';
+import { floor } from './utils/transformer.js';
+import { validateCalculator } from './utils/validateCalculator.js';
+
+const calculateWithValidation = (operator, transformer = Number) => {
+  return (...operands) => {
+    try {
+      if (validateCalculator(operator, ...operands)) {
+        return transformer(operator(...operands));
+      }
+    } catch (error) {
+      throw new Error(error);
+    }
+  };
+};
```

Comment:
```suggestion
const calculateWithValidation = (operator, transformer = Number) => {
  return (...operands) => {
    if (validateCalculator(operator, ...operands)) {
      return transformer(operator(...operands));
    }
  };
};
```
에러가 발생하자마자 어차피 catch로 진입하게 되고, 다시 여기서 throw를 하고 있기 때문에, try/catch 자체가 이 상황에선 불필요하답니다 ㅎㅎ